cmd/gke-exec-auth-plugin/cache.go

```diff
@@ -128,15 +134,18 @@ func getExistingKeyCert(dir string) ([]byte, []byte, bool) {
 		return nil, nil, false
 	}
 	age := time.Now().Sub(parsedCert.NotBefore)
+	remaining := parsedCert.NotAfter.Sub(time.Now())
+	// Note: case order matters. Always check outside of expiry bounds first
+	// and put cases that return non-nil key/cert at the bottom.
 	switch {
+	case remaining < responseExpiry:
+		glog.Infof("existing cert expired or will expire in <%v, requesting new one", responseExpiry)
+		return nil, nil, false
 	case age < 0:
 		glog.Warningf("existing cert not valid yet, requesting new one")
```

Comment:
Should we add e.g. 5 minutes of leeway on this check? I'm worried about clock skew.